2-classes-db/task.md

```diff
@@ -1,79 +1,114 @@
 
 ```
+
 ### 2. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose description includes the word 'fee'.
-```sql
 
+```sql
+SELECT date, transaction_no, supplier_inv_no,description, amount FROM spends WHERE description LIKE '%fee%';
 ```
+
 ### 3. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose description includes the word 'Fee'.
-```sql
 
+```sql
+SELECT date, transaction_no, supplier_inv_no,description, amount FROM spends WHERE description LIKE '%Fee%';
 ```
+
 ### 4. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose description includes the word 'fee' (case insensitive). You will need to search 'https://www.postgresql.org/docs/' to solve this.
-```sql
 
+```sql
+SELECT date, transaction_no, supplier_inv_no,description, amount FROM spends WHERE description ILIKE '%fee%';
 ```
+
 ### 5. Show the date, transaction_no, supplier_inv_no, description and amount for those transactions whose amount is £25,000, £30,000, £35,000 or £40,000.
-```sql
 
+```sql
+SELECT date, transaction_no, supplier_inv_no, description, amount FROM spends WHERE amount IN (25000,30000,35000,40000);
 ```
+
 ### 6. Show the date, the supplier_id, the description and the amount for transactions with the expense area of 'Better Hospital Food'. You could do a query to get the expense_area_id first then do a query to find the dates, supplier_ids and amounts. But it would be better to do this all in one query by linking the tables together using INNER JOINs.
-```sql
 
+```sql
+SELECT spends.date, spends.supplier_id, spends .description, spends .amount FROM spends  INNER JOIN expense_areas  ON spends.expense_area_id = expense_areas.id  WHERE expense_area = 'Better Hospital Food';
 ```
+
 ### 7. Show the date, supplier name, description and amount for transactions with the expense area of 'Better Hospital Food'. You will need to INNER JOIN another table to be able to do this.
-```sql
 
+```sql
+SELECT date, supplier, description, amount
+FROM spends
+INNER JOIN suppliers ON spends.supplier_id = suppliers.id
+INNER JOIN expense_areas ON spends.expense_area_id = expense_areas.id
+WHERE expense_area='Better Hospital Food';
 ```
+
 ### 8. We have just received a late invoice for April! Add a new row to the spends table:
+
     dated 1st April 2021
     with a description of 'Computer Hardware Dell'
     transaction number is 38104091 and the supplier's inv no is '3780119655'
     the supplier is 'COMPUTACENTER (UK) LTD' (id 16)
     the expense type is 'Computer Hardware Purch' (id 7)
     the expense area is 'ICT Contingency' (id 18)
     for £32,000.
-```sql
 
+```sql
+INSERT INTO spends (expense_type_id, expense_area_id, supplier_id,date,transaction_no,supplier_inv_no,description, amount)
+VALUES(7,18,16, '2021-04-01',38104091,3780119655, 'Computer Hardware Dell', 32000);
```

Comment:
CORRECT - However, depending on the parser being used, the transaction number may not work unless it is enclosed in quotes, in line with the defined data type.